### ISSUE TITLE
Converge error: Malformed response first line

### ISSUE THREAD

**Issue by tf** [issuu/sure-deploy#7]
No matter if I run `sure-deploy` from the docker image or as a manually compiled executable, I always get:

```
$ sure-deploy converge --host <ip> --port <port> <stack_name>
(monitor.ml.Error
 (Failure "Malformed response first line: \021\003\001\000\002\002")
 ("Raised at file \"pervasives.ml\", line 32, characters 17-33"
  "Called from file \"src/deferred1.ml\", line 20, characters 40-45"
  "Called from file \"src/job_queue.ml\", line 159, characters 6-47"))

$ docker -v
Docker version 18.05.0-ce, build f150324
```
On Ubuntu 16.04.

Any hints appreciated.

**Comment by Leonidas-from-XIV** [comment]
Sorry, I haven't seen your message for some reason.

Did you install the `tls` or `ssl` package? This looks like it is getting binary data, where it expects JSON.

**Comment by Leonidas-from-XIV** [comment]
Also, can you try the docker-image (`leonidasfromxiv/sure-deploy:49`)? Does this one work for you?

**Comment by tf** [comment]
Unfortunately this leads to a similar result:
```
# docker run --rm -ti leonidasfromxiv/sure-deploy:49 converge --host <ip> --port <port> <stack> 
(monitor.ml.Error
 (Failure "Malformed response first line: \021\003\001\000\002\002")
 ("Raised at file \"pervasives.ml\", line 32, characters 17-33"
  "Called from file \"src/deferred1.ml\", line 20, characters 40-45"
  "Called from file \"src/job_queue.ml\", line 159, characters 6-47"))
```

**Comment by tf** [comment]
> Did you install the tls or ssl package

This should not be required when using the Docker image right?

**Comment by Leonidas-from-XIV** [comment]
> No. But this should not be required when using the Docker image right?

No, it shouldn't be required.

Could you try

```
curl -v <host>:<port>/v1.24/services
```

Does this output a giant JSON file? Does it pick SSL/TLS?

**Comment by tf** [comment]
Ah, I wasn't aware that `sure-deploy` requires a TCP connection to the Docker daemon. I just tried running the command on the swarm manager and expected it to work. I need the TLS certificate setup, right?

**Comment by Leonidas-from-XIV** [comment]
Yes, it does HTTP calls to the API that Docker Swarm exposes. Ours is internal and does not have TLS, so I never tested it this way. The OPAM file does not depend on SSL so the docker image does not support it.

I think if you install the `tls` or `ssl` OPAM packages (which then lead to a recompile of `cohttp`, the HTTP library used) and rebuild sure-deploy it might work for you.

**Comment by n1ru4l** [comment]
Some info regarding TLS

https://github.com/mirage/ocaml-cohttp/issues/479
